src/HooglePlus/FilterTest.hs

```diff
@@ -105,39 +121,29 @@ mapRandomParamsList imports env count = do
                     _ -> list
 
 -- generate test input as line of code
-generateTestInput :: [String] -> FunctionSigniture -> IO String
+generateTestInput :: [String] -> FunctionSignature -> IO String
 generateTestInput imports env = do
   result <- generateTestInput' imports env
   return $ unwords (map snd result)
 
 -- generate test input as pair (type, value)
-generateTestInput' :: [String] -> FunctionSigniture -> IO [(String, String)]
-generateTestInput' imports (constraints, args, retType) = mapM f (transformType constraints args)
+generateTestInput' :: [String] -> FunctionSignature -> IO [(String, String)]
+generateTestInput' imports funcSig = mapM f (_argsType funcSig)
   where
     f x = let str = show x in do
       val <- generateRandomValue imports str
       return (str, val)
 
-    transformType [] args = args
-    transformType (envItem:env) args = transformType env args'
-      where
-        args' = map (replace envItem) args
-
-        -- replace all occurrance of nameL to its instantiated type
-        replace (Polymorphic nameL, constraint) (Polymorphic nameR) =
-          if nameL == nameR then
-            case constraint of
-              "Eq" -> Concrete "Int"
-              "Num" -> Concrete "Int"
-          else Polymorphic nameR
-        replace constraint (ArgTypeList argType) = ArgTypeList (replace constraint argType)
-        replace _ x = x
-
 eval_ :: [String] -> String -> Int -> IO (Maybe (Either InterpreterError String))
 eval_ modules line time = timeout time $ runInterpreter $ do
   setImports modules
   eval line
 
+runStmt_ :: [String] -> String -> Int -> IO (Maybe (Either InterpreterError ()))
```

Comment:
This type looks like it should just be `Maybe InterpreterError` instead of this. Does it encode any more information--what is meant by `Nothing :: Maybe (Either Error ())` here?

Reply:
Nvm, it encodes a timeout in the nothing case. It's fine.